src/Compilation.zig

```diff
@@ -4985,6 +5025,49 @@ pub fn lockAndSetMiscFailure(
     return setMiscFailure(comp, tag, format, args);
 }
 
+fn parseLddStderr(comp: *Compilation, comptime prefix: []const u8, stderr: []const u8) Allocator.Error!void {
+    var context_lines = std.ArrayList([]const u8).init(comp.gpa);
+    defer context_lines.deinit();
+
+    var current_err: ?*LldError = null;
+    var lines = mem.split(u8, stderr, std.cstr.line_sep);
+    while (lines.next()) |line| {
+        if (mem.startsWith(u8, line, prefix ++ ":")) {
+            if (current_err) |err| {
+                err.context_lines = context_lines.toOwnedSlice();
+            }
+
+            const duped_msg = try comp.gpa.dupe(u8, line);
+            errdefer comp.gpa.free(duped_msg);
+
+            current_err = try comp.lld_errors.addOne(comp.gpa);
+            current_err.?.* = .{ .msg = duped_msg };
+        } else if (current_err != null) {
+            const context_prefix = ">>> ";
+            var trimmed = mem.trimRight(u8, line, &std.ascii.whitespace);
+            if (mem.startsWith(u8, trimmed, context_prefix)) {
+                trimmed = trimmed[context_prefix.len..];
+            }
```

Comment:
I think this could be trimmed from both sides since the error notes are already indented but if you want to preserve the errors in their original form I'm OK with that too.
```suggestion
            var trimmed = line;
            if (mem.startsWith(u8, line, context_prefix)) {
                trimmed = mem.trim(u8, trimmed[context_prefix.len..], &std.ascii.whitespace);
            }
```

Reply:
I think preserving the beginning whitespace is nicer for this case, to maintain alignment:

```
error: lld-link: undefined symbol: definitely_not_existing
    note: referenced by C:\cygwin64\home\kcbanner\temp\zig_link_problem\src\main.zig:6
    note:               C:\cygwin64\home\kcbanner\temp\zig_link_problem\zig-cache\o\90ba81227403799d17d7062446163b66\zig_link_problem.exe.obj:(main.main)
```